VisualStudio/src/CookingPatches.cs

```diff
@@ -10,19 +10,26 @@ internal class CookingPotItem_DoSpecialActionFromInspectMode
         internal static bool Prefix(CookingPotItem __instance)
         {
             float waterAmount = __instance.m_LitersWaterBeingBoiled;
-            if (__instance.GetCookingState() == CookingPotItem.CookingState.Ready && waterAmount > 0) //patch does not apply while cooking
+            if (waterAmount <= 0) //only applies with water 
+            {
+                return true;
+            }
+            bool is_ready = __instance.GetCookingState() == CookingPotItem.CookingState.Ready;
+            bool is_not_ready_and_no_fire = __instance.GetCookingState() == CookingPotItem.CookingState.Cooking && !__instance.AttachedFireIsBurning();
+            if (is_ready || is_not_ready_and_no_fire) //patch applies if ready or if still cooking but no fire.
             {
                 GearItem gearItem = __instance.GetComponent<GearItem>();
                 WaterSupply waterSupply = gearItem.m_WaterSupply;
-                if (waterSupply == null)//if it doesn't exist
+                if (waterSupply == null)
                 {
-                    waterSupply = gearItem.gameObject.AddComponent<WaterSupply>();//create one
-                    gearItem.m_WaterSupply = waterSupply;//and assign it
-                }
+                    waterSupply = gearItem.gameObject.AddComponent<WaterSupply>();
+                    gearItem.m_WaterSupply = waterSupply;
+                } 
```

Comment:
These are in my opinion unnecessary comments and clutter the code. Such comments should be used on ambiguous parts. (Yes, I didn't do it :D )